pkg/actuators/machine/validations_test.go

```diff
@@ -71,16 +71,14 @@ func TestValidateMachine(t *testing.T) {
 			expectIsValid: false,
 		},
 		{
-			// This test now fails due to duplicate check of MemoryMB == 0 that happens in validateGuaranteedMemory func
-			// in validations.go, setting expectIsValid: false instead of true before it's fixed
 			name: "validation of machine provider spec with only instance type ID succeeds",
 			spec: BasicValidSpec(func(omps *v1beta1.OvirtMachineProviderSpec) *v1beta1.OvirtMachineProviderSpec {
 				omps.CPU = nil
 				omps.MemoryMB = 0
 				omps.InstanceTypeId = "Metal"
 				return omps
 			}),
-			expectIsValid: false,
+			expectIsValid: true,
```

Comment:
I want to test this: https://github.com/openshift/cluster-api-provider-ovirt/blob/master/pkg/actuators/machine/validations.go#L67

To be more concrete, this test validates machine provider spec where InstanceTypeID is specified. And InstanceTypeID cannot be specified at the same time as CPU or MemoryMB. The BasicValidSpec we're using  as basis for the validations tests has both CPU and MemoryMB set, so I'm "unsetting" them, and then setting InstanceTypeId - a configuration that should succeed. 

But right now it's failing due to duplicate check if MemoryMB == 0 (I set the expectIsValid to false temporarily for this test after being advised to do so by Michael, so the tests could be merged, but it should be expectIsValid: true). 

The duplicate check I'm talking about is that both here: 
https://github.com/openshift/cluster-api-provider-ovirt/blob/master/pkg/actuators/machine/validations.go#L75 

and here: 
https://github.com/openshift/cluster-api-provider-ovirt/blob/master/pkg/actuators/machine/validations.go#L121 

we're checking the same thing - if MemoryMB == 0, and in the second case it always checks for this. But since, when we're setting InstanceTypeId, we cannot set MemoryMB, it shouldn't want this condition, and therefore it's a bug. 
That, in my opinion, is most easily rectified by removing the "duplicate check". 

Does my thinking make sense?